### PR TITLE
DOC: update the MAPIE homepage

### LINKED ISSUE #335: Improve the MAPIE homepage
Right now, the MAPIE homepage does not really reflect the full library.

### PR DESCRIPTION
**Pull request by thibaultcordier**
# Description

Propose to update the MAPIE homepage to reflect the entire library as well as the latest features.

Fixes #335

## Type of change

- This change requires a documentation update

# Checklist

- [x] I have read the [contributing guidelines](https://github.com/simai-ml/MAPIE/blob/master/CONTRIBUTING.rst)
- [x] I have updated the [HISTORY.rst](https://github.com/simai-ml/MAPIE/blob/master/HISTORY.rst) and [AUTHORS.rst](https://github.com/simai-ml/MAPIE/blob/master/AUTHORS.rst) files
- [x] Linting passes successfully : `make lint`
- [x] Typing passes successfully : `make type-check`
- [x] Unit tests pass successfully : `make tests`
- [x] Coverage is 100% : `make coverage`
- [x] Documentation builds successfully : `make doc`

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -41,13 +41,32 @@
...
+Importantly, **MAPIE** contributes to the wide diffusion of the attractive **Conformal Prediction** (CP) framework for regression
+and classification settings that is model and use case agnostic with mathematical guarantees on the marginal coverages on the prediction sets
+with few assumptions (distribution-free and data-exchangeability assumptions) [1-2].
+
+Prediction sets output by **MAPIE** encompass both aleatoric and epistemic uncertainties
+and are backed by strong theoretical guarantees using a variety of conformal prediction methods [3-9]
+and a variaty of conformal risk control methods [10-12].
 
 
 🔗 Requirements
```
Comment (LacombeLouis) [inline README.rst:72]: Why don't we add something about the predictions set for classification?

File: README.rst [hunk 2]
```diff
@@ -41,13 +41,32 @@
...
-using your favourite scikit-learn-compatible model for
-single-output regression or multi-class classification settings.
+Quantifying the uncertainties and controlling the risks of ML model predictions is of crucial importance
+for developing and deploying reliable artificial intelligence (AI) systems. Uncertainty quantification (UQ)
+involves all the stakeholders who develop and use AI models.
 
-Prediction intervals output by **MAPIE** encompass both aleatoric and epistemic
-uncertainties and are backed by strong theoretical guarantees thanks to conformal
-prediction methods [1-7].
+**MAPIE** is an open-source Python library hosted on scikit-learn-contrib project that allows you to:
```
Comment (LacombeLouis) [inline README.rst:48]: Maybe we could add links to the places in the documentations where we "wrap your favorite scikit-learn-compatible model".

**Review by LacombeLouis — APPROVED** [review]
Thank you for updating the page. I would add links to the calibration page/favorite wrapper. Additionally, I think it would be good to mention right now the classification and show an example in the home page. It's a key usage of MAPIE now.

**Comment by LacombeLouis** [comment]
Also, I'm not sure if we still need the part about split vs. CV. We explain this later, and I don't think it helps make a friendly home page.

File: README.rst (hunk 2 above)
Comment (thibaultcordier) [reply README.rst:48]: I will make a proposition.

File: README.rst [hunk 3]
```diff
@@ -41,13 +41,32 @@
...
+  with as little assumption on the data or the model as possible.
+- Implemented methods follow **state-of-the-art trends** that respect programming standards in order to develop trustworthy AI systems.
+
+Importantly, **MAPIE** contributes to the wide diffusion of the attractive **Conformal Prediction** (CP) framework for regression
+and classification settings that is model and use case agnostic with mathematical guarantees on the marginal coverages on the prediction sets
+with few assumptions (distribution-free and data-exchangeability assumptions) [1-2].
+
+Prediction sets output by **MAPIE** encompass both aleatoric and epistemic uncertainties
+and are backed by strong theoretical guarantees using a variety of conformal prediction methods [3-9]
+and a variaty of conformal risk control methods [10-12].
```
[the remark this answers is not in the document]
Comment (thibaultcordier) [reply README.rst:71]: I will make a proposition.

File: README.rst (hunk 1 above)
Comment (thibaultcordier) [reply README.rst:72]: I will make a proposition.

File: README.rst (hunk 3 above)
[the remark this answers is not in the document]
Comment (thibaultcordier) [reply README.rst:71]: I merge them in one code block.

File: README.rst (hunk 2 above)
Comment (thibaultcordier) [reply README.rst:48]: I try to add a hyperlink in the file.

**Review by thibaultcordier — COMMENTED** [review]
I take account the comments of @LacombeLouis, and give some suggestion.

File: README.rst [hunk 4]
```diff
@@ -41,49 +41,150 @@
...
+    from mapie.regression import MapieRegressor
+    mapie_regressor = MapieRegressor(estimator=regressor, method='plus', cv=5)
+
+.. code:: python
+
+    # Uncertainty quantification for classification problem
+    from mapie.classification import MapieClassifier
+    mapie_classifier = MapieClassifier(estimator=classifier, method='score', cv=5)
+
+.. code:: python
```
Comment (LacombeLouis) [inline README.rst:71]: I would add multi-label. I think let's keep it to two simple use-cases. Regression and classification?

**Review by LacombeLouis — CHANGES_REQUESTED** [review]
I think this is better. But too lengthy. I would keep it to only regression and classification (and no need to initialize the estimators, let the default ones from MAPIE?)

I'm not sure if it won't be too long still then. In that case, I would get rid of the examples showing the split and CV.

File: README.rst [hunk 5]
```diff
@@ -157,11 +261,69 @@ It is given by the alpha parameter defined in ``MapieRegressor``, here equal to
...
+    plt.scatter(X[:, 0], X[:, 1], c=y, cmap='tab20c')
+    plt.xlabel("x1")
+    plt.ylabel("x2")
+    plt.title(
+        f"Target and effective coverages for "
+        f"alpha={alpha[0]:.2f}: ({1-alpha[0]:.3f}, {coverage_scores[0]:.3f})"
+    )
+    plt.show()
+
+.. image:: https://github.com/scikit-learn-contrib/MAPIE/raw/master/doc/images/quickstart_2.png
```
Comment (LacombeLouis) [inline README.rst:324]: Doesn't show well.
Screenshot 2023-09-06 at 17 42 35

File: doc/split_cross_conformal.rst [hunk 6]
```diff
@@ -0,0 +1,29 @@
+#########################################
+Split/Cross-Conformal Prediction in MAPIE
+#########################################
+
+**MAPIE** is basically based on two types of techniques:
+
+1. Cross conformal predictions
```
Comment (LacombeLouis) [inline doc/split_cross_conformal.rst:7]: I would put split first.

**Review by LacombeLouis — APPROVED** [review]
Perfect for me! 👍🏼 

Could you not use the newest diagrams you made for CV and split?

File: README.rst [hunk 7]
```diff
@@ -41,159 +41,115 @@
...
-**MAPIE** allows you to easily estimate prediction intervals (or prediction sets)
-using your favourite scikit-learn-compatible model for
-single-output regression or multi-class classification settings.
+**MAPIE** is an open-source Python library for quantifying uncertainties and controlling the risks of machine learning models.
+It is a scikit-learn-contrib project that allows you to:
 
-Prediction intervals output by **MAPIE** encompass both aleatoric and epistemic
-uncertainties and are backed by strong theoretical guarantees thanks to conformal
-prediction methods [1-7].
+- easily **compute conformal prediction intervals** (or prediction sets) with controlled (ou guaranteed) marginal coverage rate
```
Comment (vincentblot28) [inline README.rst:47]: or instead of ou

File: README.rst [hunk 8]
```diff
@@ -41,159 +41,115 @@
...
+  for regression [3,4,8], classification (binary and multi-class) [5-7] and time series [9].
+- easily **control risks** of more complex tasks such as multi-label classification,
+  segmentation in computer vision (probabilistic guarantees on recall, precision, ...) [10-12].
```
Comment (vincentblot28) [inline README.rst:50]:
```suggestion
  semantic segmentation in computer vision (probabilistic guarantees on recall, precision, ...) [10-12].
```

File: README.rst [hunk 9]
```diff
@@ -41,159 +41,115 @@
...
+- easily **wrap your favorite scikit-learn-compatible model** for the purposes just mentioned.
```
Comment (vincentblot28) [inline README.rst:51]:
```suggestion
- easily **wrap your favorite scikit-learn-compatible model or any model wrapped with a scikit-learn-compatible class** for the purposes just mentioned.
```

File: README.rst [hunk 10]
```diff
@@ -41,159 +41,115 @@
...
 
-.. code:: sh
 
-    $ pip install mapie
+🔗 Requirements
+===============
 
-or via `conda`:
+- **MAPIE** runs on Python 3.7+.
+- **MAPIE** stands on the shoulders of giants. Its only internal dependencies are `scikit-learn <https://scikit-learn.org/stable/>`_ and `numpy=>1.21 <https://numpy.org/>`_.
```
Comment (vincentblot28) [inline README.rst:81]: don't we have scipy also now ?

File: README.rst (hunk 7 above)
Comment (vincentblot28) [inline README.rst:47]:
```suggestion
- Easily **compute conformal prediction intervals** (or prediction sets) with controlled (ou guaranteed) marginal coverage rate
```

File: README.rst [hunk 11]
```diff
@@ -41,159 +41,115 @@
...
-single-output regression or multi-class classification settings.
+**MAPIE** is an open-source Python library for quantifying uncertainties and controlling the risks of machine learning models.
+It is a scikit-learn-contrib project that allows you to:
 
-Prediction intervals output by **MAPIE** encompass both aleatoric and epistemic
-uncertainties and are backed by strong theoretical guarantees thanks to conformal
-prediction methods [1-7].
+- easily **compute conformal prediction intervals** (or prediction sets) with controlled (ou guaranteed) marginal coverage rate
+  for regression [3,4,8], classification (binary and multi-class) [5-7] and time series [9].
+- easily **control risks** of more complex tasks such as multi-label classification,
```
Comment (vincentblot28) [inline README.rst:49]:
```suggestion
- Easily **control risks** of more complex tasks such as multi-label classification,
```

File: README.rst [hunk 12]
```diff
@@ -41,159 +41,115 @@
...
-Its only internal dependencies are `scikit-learn <https://scikit-learn.org/stable/>`_ and `numpy=>1.21 <https://numpy.org/>`_.
+    # Uncertainty quantification for classification problem
+    from mapie.classification import MapieClassifier
+    mapie_classifier = MapieClassifier(estimator=classifier, method='score', cv=5)
 
+Implemented methods in **MAPIE** respect three fundamental pillars:
 
-🛠 Installation
-===============
+- they are **model and use case agnostic**,
```
Comment (vincentblot28) [inline README.rst:70]:
```suggestion
- They are **model and use case agnostic**, 
```

File: README.rst [hunk 13]
```diff
@@ -41,159 +41,115 @@
...
+    # Uncertainty quantification for classification problem
+    from mapie.classification import MapieClassifier
+    mapie_classifier = MapieClassifier(estimator=classifier, method='score', cv=5)
 
+Implemented methods in **MAPIE** respect three fundamental pillars:
 
-🛠 Installation
-===============
+- they are **model and use case agnostic**, 
+- they possess **theoretical guarantees** under minimal assumptions on the data and the model,
```
Comment (vincentblot28) [inline README.rst:71]:
```suggestion
- They possess **theoretical guarantees** under minimal assumptions on the data and the model,
```

File: README.rst [hunk 14]
```diff
@@ -41,159 +41,115 @@
...
+- they are based on **peer-reviewed algorithms** and respect programming standards.
```
Comment (vincentblot28) [inline README.rst:72]:
```suggestion
- They are based on **peer-reviewed algorithms** and respect programming standards.
```

File: README.rst [hunk 15]
```diff
@@ -41,159 +41,115 @@
...
+⚡ Quickstart
+=============
+
+Here we propose two basic uncertainty quantification problems for regression and classification tasks with scikit-learn.
+
+As **MAPIE** is compatible with the standard scikit-learn API, you can see that with just these few lines of code:
 
-Let us start with a basic regression problem. 
-Here, we generate one-dimensional noisy data that we fit with a linear model.
+- how easy it is **to wrap your favorite scikit-learn-compatible model** around your model.
```
Comment (vincentblot28) [inline README.rst:103]:
```suggestion
- How easy it is **to wrap your favorite scikit-learn-compatible model** around your model.
```

File: README.rst [hunk 16]
```diff
@@ -41,159 +41,115 @@
...
+- how easy it is **to follow the standard sequential** ``fit`` and ``predict`` process like any scikit-learn estimator.
```
Comment (vincentblot28) [inline README.rst:104]:
```suggestion
- How easy it is **to follow the standard sequential** ``fit`` and ``predict`` process like any scikit-learn estimator.
```

File: README.rst [hunk 17]
```diff
@@ -41,159 +41,116 @@
...
+It is a scikit-learn-contrib project that allows you to:
 
-Prediction intervals output by **MAPIE** encompass both aleatoric and epistemic
-uncertainties and are backed by strong theoretical guarantees thanks to conformal
-prediction methods [1-7].
+- Easily **compute conformal prediction intervals** (or prediction sets) with controlled (or guaranteed) marginal coverage rate
+  for regression [3,4,8], classification (binary and multi-class) [5-7] and time series [9].
+- Easily **control risks** of more complex tasks such as multi-label classification,
+  semantic segmentation in computer vision (probabilistic guarantees on recall, precision, ...) [10-12].
+- Easily **wrap your favorite scikit-learn-compatible model or any model wrapped with a scikit-learn-compatible class**
```
Comment (vincentblot28) [inline README.rst:51]:
```suggestion
- Easily **wrap any model (scikit-learn, tensorflow, pytorch, ...) with, if needed, a scikit-learn-compatible wrapper**
```

**Review by vincentblot28 — APPROVED** [review]

**Pull request merged**